src/include/reservation.h

```diff
@@ -330,10 +317,13 @@ struct resc_resv {
 #define	Q_CHNG_ENABLE		0
 #define	Q_CHNG_START		1
 
-extern resc_resv  *find_resv(char *);
-extern resc_resv  *resc_resv_alloc(void);
-extern void  resv_purge(resc_resv *);
-extern int   start_end_dur_wall(void *, int);
+extern AVL_IX_DESC *AVL_resvs_byid;
+extern AVL_IX_DESC *AVL_resvs_byqn;
```

Comment:
I don't know much about reservations internal structure, but there were two find_resv_* calls and both were doing lookup with different parameters + reservation structure have both parameters defined structure so I thought both are different and we need different AVL tree based on those parameters. If you are sure that id and queue name will be always the same then I can remove on AVL tree.

Reply:
So the reservations usually have an id like <resv-number>.<server-name>@<server>
As long as you use only <resv-number> it should be ok.
Naming the reservation queue as the reservation ID is something that hasn't changed. I'd recommend you make it one single tree.

Reply:
Does that mean we don't need two find_resv_* functions also? Can I remove one (like find_resv_by_quename()) and use find_resv_byid() instead?

Reply:
I'd say keep the two functions. They give a logical separation for someone who is coding. Internally find_resv_by_quename could call find_resv_byid.
What do you think?

Reply:
Honestly, for me, it will be more confusing if we have two functions doing exactly the same functionality then why we need two functions at all? otherwise, 2nd function will be redundant.
Instead, I would say, remove find_resv_by_quename() and use find_resv() (renamed version of find_resv_byid) everywhere.

Reply:
I’m of the opinion that it’s ok to have two different functions. How they are implemented is internal to function. As for the caller it will be coming from two different call paths so it will be nice to have distinct functions.

Reply:
@bhroam, @subhasisb and others, I would like to know your opinion on this, please.

Reply:
if the two functions are going to do the exact same thing - I would remove one of them.

Reply:
Since there is two of you sharing the same opinion, I’m fine having one function. Just make it so that the function name is relevant in both contexts.

Reply:
I guess my opinion depends on what the arguments to the two functions are.  If they are both strings, then, yes, they can be combined to search for the short form of a reservation.  There is no difference between the short form (R1234) and the long form (R1234.server) of a reservation, since you can't have reservations from other servers.  If the queue based one takes a queue pointer, then it's different, and I can see the point of it.  Of course to use an AVL tree, you need string keys, so I suspect it is the former.

Reply:
Yes @bhroam both functions have a string as an argument.

Reply:
Then I agree that we should only have one.  Of course I'm not sure we should assume that the reservation queue name will always be the same as its ID.  It is currently that, but I have heard ideas over the years of making the queue name customizable.  I don't know how serious they are though.

Reply:
> Of course I'm not sure we should assume that the reservation queue name will always be the same as its ID.

So as per below (thanks to @arungrover for pointing it out), It looks like we will always have reservation queue name == reservation ID
https://github.com/PBSPro/pbspro/blob/013503b6d63c80a8d691a30a207de7cb6f16cb87/src/server/req_rescq.c#L839-L849

Reply:
So in today's context, both the queue name and reservation name are the same. I guess in future if that changes then we will have a need to create another tree with queue names as the key.